### PR TITLE
[Merged by Bors] - feat: `(a • b) * c • d = (a * c) • (b * d)`

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by b-mehta — COMMENTED** [review]
bors merge

File: Mathlib/Algebra/Group/Action/Defs.lean [hunk 1]
```diff
@@ -354,6 +354,16 @@ lemma smul_smul_smul_comm [SMul α β] [SMul α γ] [SMul β δ] [SMul α δ] [S
     [IsScalarTower α β δ] [IsScalarTower α γ δ] [SMulCommClass β γ δ] (a : α) (b : β) (c : γ)
     (d : δ) : (a • b) • c • d = (a • c) • b • d := by rw [smul_assoc, smul_assoc, smul_comm b]
 
+@[to_additive]
+lemma smul_mul_smul_comm [Mul α] [Mul β] [SMul α β] [IsScalarTower α β β]
+    [IsScalarTower α α β] [SMulCommClass β α β] (a : α) (b : β) (c : α) (d : β) :
+    (a • b) * c • d = (a * c) • (b * d) := smul_smul_smul_comm a b c d
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Action/Defs.lean:360]: for clarity:
```suggestion
    (a • b) * (c • d) = (a * c) • (b * d) := smul_smul_smul_comm a b c d
```

File: Mathlib/Algebra/Group/Action/Defs.lean [hunk 2]
```diff
@@ -354,6 +354,16 @@ lemma smul_smul_smul_comm [SMul α β] [SMul α γ] [SMul β δ] [SMul α δ] [S
...
+
+@[to_additive]
+lemma mul_smul_mul_comm [Mul α] [Mul β] [SMul α β] [IsScalarTower α β β]
+    [IsScalarTower α α β] [SMulCommClass α β β] (a b : α) (c d : β) :
+    (a * b) • (c * d) =  (a • c) * b • d := smul_smul_smul_comm a b c d
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Action/Defs.lean:365]:
```suggestion
    (a * b) • (c * d) = (a • c) * (b • d) := smul_smul_smul_comm a b c d
```
This is just `smul_mul_smul_comm` backwards, right? Happy to keep it anyway.

File: Mathlib/Algebra/Group/Action/Defs.lean [hunk 3]
```diff
@@ -354,6 +354,16 @@ lemma smul_smul_smul_comm [SMul α β] [SMul α γ] [SMul β δ] [SMul α δ] [S
     [IsScalarTower α β δ] [IsScalarTower α γ δ] [SMulCommClass β γ δ] (a : α) (b : β) (c : γ)
     (d : δ) : (a • b) • c • d = (a • c) • b • d := by rw [smul_assoc, smul_assoc, smul_comm b]
 
+@[to_additive]
+lemma smul_mul_smul_comm [Mul α] [Mul β] [SMul α β] [IsScalarTower α β β]
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Action/Defs.lean:358]: How is this different from the existing `smul_mul_smul`?

Comment (YaelDillies) [reply Mathlib/Algebra/Group/Action/Defs.lean:358]: It is more general

Comment (eric-wieser) [reply Mathlib/Algebra/Group/Action/Defs.lean:358]: (deprecating that lemma, if they're truly the same, sounds find to me)

File: Mathlib/Algebra/Group/Action/Defs.lean [hunk 4]
```diff
@@ -444,19 +458,12 @@ instance IsScalarTower.left : IsScalarTower M M α where
 
 variable {M}
 
-/-- Note that the `IsScalarTower M α α` and `SMulCommClass M α α` typeclass arguments are
-usually satisfied by `Algebra M α`. -/
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Action/Defs.lean:448]: I think this comment should be preserved.

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**